src/components/Paper/components/Palette/index.js

```diff
@@ -25,18 +33,86 @@ function Palette(props) {
     }
   }
 
+  // change in custom color palette menu
+  const handlePaletteChange = (color) => {
+    setPaletteColor(color.hex);
+  }
+
+  // change in custom color palette menu completed
+  const handleChangeComplete = (color) => {
+    setPaletteColor(color.hex);
+    props.onSelectColor(color.hex);
+  }
+
+  // custom color selected in top palette
+  const handleSelectCustom = () => {
+    setSelectingColor(!selectingColor);
+    props.onSelectColor(paletteColor);
+  }
+
+  // non-custom color selected in top palette
+  const handleSelectColor = (color) => {
+    setSelectingColor(false);
+    props.onSelectColor(color);
+  }
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      const selector = document.querySelector(`.${styles['color-palette__selector']}`);
+      if (selector && !selector.contains(event.target)) {
+        setSelectingColor(false);
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => document.removeEventListener('mousedown', handleClickOutside);
+  }, [setSelectingColor]);
+
   return (
-    <div className={styles['color-palette__container']}>
-      {palette.map((color) => (
+    <div>
+      <div className={styles['color-palette__container']}>
+        {palette.map((color) => (
+          <div
+            key={`color-palette__${color.substring(1)}`}
+            onClick={() => handleSelectColor(color)}
+            className={classNames(styles['color-palette__color'], {
+              [styles['color-palette__color-active']]: color === selectedColor,
+            })}
+            style={{ backgroundColor: color }}
+          />
+        ))}
         <div
-          key={`color-palette__${color.substring(1)}`}
-          onClick={() => props.onSelectColor(color)}
           className={classNames(styles['color-palette__color'], {
-            [styles['color-palette__color-active']]: color === selectedColor,
+            [styles['color-palette__color-active']]: paletteColor === selectedColor,
           })}
-          style={{ backgroundColor: color }}
-        />
-      ))}
+          style={{ backgroundColor: paletteColor }}
+          onClick={handleSelectCustom}
+        >
+        </div>
+      </div>
+      {selectingColor && 
+        <Draggable handle='.handle'>
+          <div className={styles['color-palette__selector']}>
+            <div
+              className='handle'
+              style={{backgroundColor: '#fff', borderRadius: '5px 5px 0 0', display: 'flex', justifyContent: 'space-between'}}
```

Comment:
Styling doesn't belong in JavaScript, please move this to the styles file inside this component.